app-grpc-test/src/main/java/net/spals/appbuilder/app/grpc/GrpcWebAppWithMocks.java

```diff
@@ -0,0 +1,53 @@
+package net.spals.appbuilder.app.grpc;
+
+import com.google.inject.Module;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A framework for injecting mock services into a
+ * {@link GrpcWebApp}.
+ *
+ * Example:
+ *
+ * final MyGrpcAppWithMocks myGrpcAppWithMocks = GrpcWebAppWithMocks.grpcWithMocks(new MyGrpcApp(0))
+ *     .addMockSingleton(mockedDatabase, Database.class)
+ *     .addMockSingleton(mockedNotifier, Notifier.class)
+ *
+ * GrpcTestSupport.embeddedGrpc(myGrpcAppWithMocks).before()
+ *
+ * @author tkral
+ */
+public class GrpcWebAppWithMocks extends GrpcWebApp {
+
+    private final GrpcWebApp delegateWebApp;
+    private final List<Module> mockSingletonModules = new ArrayList<>();
+
+    public static GrpcWebAppWithMocks grpcWithMocks(final GrpcWebApp grpcWebApp) {
+        return new GrpcWebAppWithMocks(grpcWebApp);
+    }
+
+    private GrpcWebAppWithMocks(
+        final GrpcWebApp delegateWebApp
+    ) {
+        super(0);
```

Comment:
Wouldn't you need to pass the delegate's port?
```suggestion
         super(delegateWebApp.getGrpcPort());
```

Reply:
Good call.